lib/wallaroo/core/sink/connector_sink/connector_sink.pony

```diff
@@ -387,25 +722,94 @@ actor ConnectorSink is Sink
   ///////////////
   // CHECKPOINTS
   ///////////////
+
   fun ref checkpoint_state(checkpoint_id: CheckpointId) =>
     """
-    ConnectorSinks don't currently write out any data as part of the checkpoint.
+    Serialize hard state (i.e., can't afford to lose it) and send
+    it to the local event log and reset 2PC state.
     """
+    if not (_twopc_state is cp.TwoPCFsmStart) then
+      @printf[I32]("2PC: ERROR: _twopc_state = %d\n".cstring(), _twopc_state())
+      Fail()
+    end
+
+    ifdef "checkpoint_trace" then
+      @printf[I32]("2PC: Checkpoint state %s at ConnectorSink %s, txn-id %s\n".cstring(), checkpoint_id.string().cstring(), _sink_id.string().cstring(), _twopc_txn_id.cstring())
+    end
+
+    let wb: Writer = wb.create()
+    wb.u64_be(_twopc_current_offset.u64())
+    wb.u64_be(_notify.acked_point_of_ref)
+    let bs = wb.done()
+
     _event_log.checkpoint_state(_sink_id, checkpoint_id,
-      recover val Array[ByteSeq] end)
+      consume bs)
 
   be prepare_for_rollback() =>
-    _prepare_for_rollback()
+    ifdef "checkpoint_trace" then
+      @printf[I32]("Prepare for checkpoint rollback at ConnectorSink %s\n".cstring(), _sink_id.string().cstring())
+    end
+    //// _clear_barriers() will put a normal message processor
```

Comment:
What is this comment referring to?

Reply:
Hm.  Would "Don't use _clear_barriers() here" be clearer, or remove the entire comment?

Reply:
If you think it's a trap someone might fall into, then that would be clearer.